src/points.rkt

```diff
@@ -279,14 +326,65 @@
              "Sampled" npts "points with exact outputs")
       (mk-pcontext (take-up-to pts (*num-points*)) (take-up-to exs (*num-points*)))]
      [else
-      (define num (max 4 (- (*num-points*) npts))) ; pad to avoid repeatedly trying to get last point
-      (debug #:from 'points #:depth 4
-             "Sampling" num "additional inputs,"
-             "on iter" num-loops "have" npts "/" (*num-points*))
-      (define pts1 (for/list ([i (in-range num)]) (sample)))
-      (define exs1 (make-exacts-halfpoints prog pts1 precondition #:log log))
-      (debug #:from 'points #:depth 4
-             "Filtering points with unrepresentable outputs")
-      (define-values (pts* exs*) (filter-p&e pts1 exs1))
-      ;; keep iterating till we get at least *num-points*
-      (loop (append pts* pts) (append exs* exs) (+ 1 num-loops))])))
+      (define num-vars (length (program-variables prog)))
+      (match precision
+        [(or 'binary64 'binary32)
+         (define num (max 4 (- (*num-points*) npts))) ; pad to avoid repeatedly trying to get last point
+         (debug #:from 'points #:depth 4
+                "Sampling" num "additional inputs,"
+                "on iter" num-loops "have" npts "/" (*num-points*))
+         (define pts1 (for/list ([n (in-range num)]) (sample)))
+         (define exs1 (make-exacts-halfpoints prog pts1 precondition))
+         (debug #:from 'points #:depth 4
+                "Filtering points with unrepresentable outputs")
+         (define-values (pts* exs*) (filter-p&e pts1 exs1))
+         ;; keep iterating till we get at least *num-points*
+         (loop (append pts* pts) (append exs* exs) (+ 1 num-loops))]
+        ['posit8
```

Comment:
Oh boy ok there's some cleanup we can do here. Looks like the main difference is in changing `sample` to `random-positX`? Also, you don't need to filter out the precondition ahead of time, `make-exacts-halfpoints` will do that for you (I think?).